src/docfx/build/xref/InternalXrefSpec.cs

```diff
@@ -61,12 +69,22 @@ public ExternalXrefSpec ToExternalXrefSpec(Context context, bool forXrefMapOutpu
                 {
                     spec.ExtensionData[key] = value.Value;
                 }
-                catch (DocfxException ex)
+
+                // circular-reference is handled in JsonSchemaTransformer
+                // no need to throw it again for xref map output
+                catch (DocfxException ex) when (forXrefMapOutput && ex.Error.Code == "circular-reference")
```

Comment:
```suggestion
                catch (Exception ex) when (DocfxException.Is(xxx))
                {
                    context.ErrorLog.Write(DeclaringFile, ex.Error);
                }
```

Reply:
Once there is always a file in the error, we do not need to rewrite it again here.